src/YesSql.Core/Serialization/DynamicJsonConverter.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace YesSql.Serialization
+{
+    public class DynamicJsonConverter : JsonConverter<object>
+    {
+        public static readonly DynamicJsonConverter Instance = new();
+
+        public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.Null:
+                    return null;
+                case JsonTokenType.False:
+                    return false;
+                case JsonTokenType.True:
+                    return true;
+                case JsonTokenType.String:
+                    return reader.GetString();
+                case JsonTokenType.Number:
+                    {
+                        if (reader.TryGetInt32(out var i))
+                            return i;
+                        if (reader.TryGetInt64(out var l))
+                            return l;
+                        // BigInteger could be added here.
```

Comment:
I was not sure that we need something here for `BigInteger`, but maybe if it fails to convert for example to a long because of the dedicated max/min values.

But we may also need to provide a `JsonBigIntegerConverter`, for info I found this one here https://github.com/graphql-dotnet/graphql-dotnet/blob/d46bde6d5e30ea1e3a314f40a408f9c86480018d/src/GraphQL.SystemTextJson/JsonConverterBigInteger.cs#L14 in which they define a `TryGetBigInteger()` that takes care if the reader has more than one segment.


    public static bool TryGetBigInteger(ref Utf8JsonReader reader, out BigInteger bi)
    {
        var byteSpan = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
        Span<char> chars = stackalloc char[byteSpan.Length];
        Encoding.UTF8.GetChars(reader.ValueSpan, chars);
        return BigInteger.TryParse(chars, NumberStyles.Integer, NumberFormatInfo.InvariantInfo, out bi);
    }